pkg/sql/conn_executor_exec.go

```diff
@@ -547,6 +547,25 @@ func (ex *connExecutor) execStmtInOpenState(
 	p.noticeSender = res
 	ih := &p.instrumentation
 
+	if maxOpen := maxOpenTransactions.Get(&ex.server.cfg.Settings.SV); maxOpen > 0 {
+		// NB: SQLTxnsOpen does not include internal executor transactions.
+		if ex.metrics.EngineMetrics.SQLTxnsOpen.Value() > maxOpen {
```

Comment:
It seems a bit odd to use reporting metrics for a guardrail, but I suppose there is no major harm in it. I guess the alternative would be adding a new atomic counter somewhere, which would basically be the same thing?